### PR TITLE
Feature/qw009 google auth

### PR DESCRIPTION
**Pull request by Misvor**

### CODE REVIEW

File: QuizWhois.Api/Controllers/AuthController.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        }
+
+        [Authorize]
+        [HttpGet("tokens")]
+        public async Task<ActionResult<string>> ShowTokens()
+        {
+            // The user is already authenticated, so this call won't trigger authentication.
+            // But it allows us to access the AuthenticateResult object that we can inspect
+            // to obtain token related values.
+            AuthenticateResult auth = await HttpContext.AuthenticateAsync();
```
Comment (sidgrouse) [inline QuizWhois.Api/Controllers/AuthController.cs:53]: lets at least call it authResult

File: QuizWhois.Api/Controllers/AuthController.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            string idToken = auth.Properties.GetTokenValue(OpenIdConnectParameterNames.IdToken);
+            string idTokenValid, idTokenIssued, idTokenExpires;
+            try
+            {
+                var payload = await GoogleJsonWebSignature.ValidateAsync(idToken);
+                idTokenValid = "true";
+                idTokenIssued = new DateTime(1970, 1, 1).AddSeconds(payload.IssuedAtTimeSeconds.Value).ToString();
```
Comment (sidgrouse) [inline QuizWhois.Api/Controllers/AuthController.cs:60]: wit, but doesn't DateTimeOffset.ToUnixTimeSeconds do the same?

File: QuizWhois.Api/Controllers/QuizController.cs [hunk 3]
```diff
@@ -19,8 +21,7 @@ public QuizController(IQuizService quizService)
         {
             _quizService = quizService;           
         }
-
-        // [Authorize]
```
Comment (sidgrouse) [inline QuizWhois.Api/Controllers/QuizController.cs:23]: why? don't we need authorization now?

File: QuizWhois.Domain/Entity/User.cs [hunk 4]
```diff
@@ -5,5 +5,7 @@ public class User
         public long Id { get; set; }
 
         public string Login { get; set; }
+
+        public string Password { get; set; }
```
Comment (sidgrouse) [inline QuizWhois.Domain/Entity/User.cs:9]: so if the db was hacked all the passwords would be compromised?

Comment (sidgrouse) [reply QuizWhois.Domain/Entity/User.cs:9]: why do we need to store it at all?

**Review by sidgrouse — CHANGES_REQUESTED** [review]

File: QuizWhois.Domain/Entity/User.cs (hunk 4 above)
Comment (Misvor) [reply QuizWhois.Domain/Entity/User.cs:9]: It is a leftover from previous solution, as for now we are not using login/password in any way, should be removed.

File: QuizWhois.Api/Controllers/AuthController.cs (hunk 1 above)
Comment (Misvor) [reply QuizWhois.Api/Controllers/AuthController.cs]: done, now it's authResult

File: QuizWhois.Api/Controllers/AuthController.cs (hunk 2 above)
Comment (Misvor) [reply QuizWhois.Api/Controllers/AuthController.cs]: Changed, now it's "DateTimeOffset.FromUnixTimeSeconds(payload.ExpirationTimeSeconds.Value).DateTime.ToString()"

**Pull request merged**